src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -976,4 +976,95 @@
       it { expect(response).to redirect_to(project_show_path(apache_maintenance_incident_project)) }
     end
   end
+
+  describe 'POST #move_path' do
+    context 'valid project' do
+      let(:repository) { create(:repository, project: apache_project) }
+      let(:path_element) { create(:path_element, repository: repository) }
+
+      context 'without direction' do
+        it 'raises an exception' do
+          expect { post :move_path, params: { project: apache_project } }.to raise_error ActionController::ParameterMissing
+        end
+      end
+
+      context 'only one path_element' do
+        let(:position) { path_element.position}
+
+        context 'direction up' do
+          before do
+            post :move_path, params: { project: apache_project, repository: repository.id, direction: 'up', path: path_element }
+          end
+
+          it { expect(path_element.reload.position).to eq(position) }
+          it { expect(flash[:notice]).to eq("Path moved up successfully") }
+          it { expect(response).to redirect_to({ action: :index, controller: :repositories, project: apache_project }) }
+        end
+
+        context 'direction down' do
+          before do
+            post :move_path, params: { project: apache_project, repository: repository.id, direction: 'down', path: path_element }
+          end
+
+          it { expect(path_element.reload.position).to eq(position) }
+          it { expect(flash[:notice]).to eq("Path moved down successfully") }
+          it { expect(response).to redirect_to({ action: :index, controller: :repositories, project: apache_project }) }
+        end
+      end
+
+      context 'three path elements' do
+        let(:path_element_2) { create(:path_element, repository: repository) }
+        let(:path_element_3) { create(:path_element, repository: repository) }
+        let(:path_elements) { [path_element, path_element_2, path_element_3] }
+
+        context 'direction up' do
+          context 'response' do
+            before do
+              post :move_path, params: { project: apache_project, repository: repository.id, direction: 'up', path: path_elements[1] }
```

Comment:
and if you remove that line https://github.com/openSUSE/open-build-service/pull/2520/files#diff-603a086da29f0e3cdbaa1d2dfc369856R1033 this is just the variable move, so if you define it up you can avoid write this twice.